consensus/scp/src/quorum_set.rs

```diff
@@ -39,18 +29,68 @@ pub enum QuorumSetMember<ID: GenericNodeId> {
     InnerSet(QuorumSet<ID>),
 }
 
-impl<
-        ID: GenericNodeId
-            + Clone
-            + Debug
-            + Display
-            + Serialize
-            + DeserializeOwned
-            + Eq
-            + PartialEq
-            + Hash,
-    > QuorumSet<ID>
-{
+impl<ID: GenericNodeId> PartialEq for QuorumSetMember<ID> {
+    fn eq(&self, other: &QuorumSetMember<ID>) -> bool {
+        match self {
+            QuorumSetMember::Node(self_node) => match other {
+                QuorumSetMember::Node(other_node) => self_node == other_node,
+                _ => false,
+            },
+            QuorumSetMember::InnerSet(self_qs) => match other {
+                QuorumSetMember::InnerSet(other_qs) => self_qs == other_qs,
+                _ => false,
+            },
+        }
+    }
+}
+impl<ID: GenericNodeId> Eq for QuorumSetMember<ID> {}
+
+impl<ID: GenericNodeId> Hash for QuorumSetMember<ID> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
```

Comment:
How is this different than what `derive(Hash)` gives us?

Reply:
we can't derive hash if we impl Eq because we have to maintain the invariant that A==B implies Hash(A)==Hash(B)... but perhaps the derived Eq will work for QuorumSetMember... my recollection was that it was problematic but I'll try again.